api/client/client.go

```diff
@@ -2277,16 +2277,35 @@ func (c *Client) GetTrustedClusters(ctx context.Context) ([]types.TrustedCluster
 }
 
 // UpsertTrustedCluster creates or updates a Trusted Cluster.
-func (c *Client) UpsertTrustedCluster(ctx context.Context, trusedCluster types.TrustedCluster) (types.TrustedCluster, error) {
-	trustedCluster, ok := trusedCluster.(*types.TrustedClusterV2)
+//
+// Deprecated: Use [Client.UpsertTrustedClusterV2] instead.
+func (c *Client) UpsertTrustedCluster(ctx context.Context, trustedCluster types.TrustedCluster) (types.TrustedCluster, error) {
+	trustedClusterV2, ok := trustedCluster.(*types.TrustedClusterV2)
 	if !ok {
-		return nil, trace.BadParameter("invalid type %T", trusedCluster)
+		return nil, trace.BadParameter("invalid type %T", trustedCluster)
 	}
-	resp, err := c.grpc.UpsertTrustedCluster(ctx, trustedCluster)
-	if err != nil {
-		return nil, trace.Wrap(err)
+	resp, err := c.grpc.UpsertTrustedCluster(ctx, trustedClusterV2)
+	return resp, trace.Wrap(err)
+}
+
+// UpsertTrustedClusterV2 creates or updates a Trusted Cluster.
+func (c *Client) UpsertTrustedClusterV2(ctx context.Context, trustedCluster types.TrustedCluster) (types.TrustedCluster, error) {
+	trustedClusterV2, ok := trustedCluster.(*types.TrustedClusterV2)
+	if !ok {
+		return nil, trace.BadParameter("invalid type %T", trustedCluster)
 	}
-	return resp, nil
+	resp, err := c.grpc.UpsertTrustedClusterV2(ctx, trustedClusterV2)
+	if trace.IsNotImplemented(err) {
+		// Try to print a nicer error message when newer clients connect to
+		// older auth servers that don't recognize the new gRPC.
+		authVersion := "unknown"
+		if pingResp, err := c.Ping(ctx); err == nil && pingResp.ServerVersion != "" {
+			authVersion = pingResp.ServerVersion
+		}
+		return resp, trace.Wrap(err, "client version (%s) is likely newer than your auth server version (%s), "+
+			"consider upgrading your auth server", api.Version, authVersion)
```

Comment:
Do we really want to advise upgrading the auth like that? If this error message started appearing in some automation and SREs were scrambling for a fix, surely they'd rather downgrade the operator or tf provider rather than upgrade the cluster?

Reply:
If the operator receives a `NotImplemented` error, the only solution would be to update the Auth Service to a version that supports this new endpoint.

This shouldn't be a problem though, the teleport operator does not support trusted_clusters yet so it won't break any existing automation.